docs/articles/basics/first_bot.md

```diff
@@ -0,0 +1,221 @@
+`DSharpPlus.VoiceNext`|Add-on which enables connectivity to Discord voice channels.
+`DSharpPlus.Rest`|REST-only Discord client.
+
+<br/>
+We'll only need the `DSharpPlus` package for the basic bot we'll be writing in this article.<br/>
+Select it from the list then click the `Install` button to the right.
+
+![Install DSharpPlus](/images/basics_first_bot_08.png)
+
+You're now ready to write some code!
+
+
+## First Lines of Code
+DSharpPlus implements [Task-based Asynchronous Pattern](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/consuming-the-task-based-asynchronous-pattern).
+Because of this, the majority of DSharpPlus methods must be executed in a method marked as `async` so they can be properly `await`ed.
+
+Due to the way the compiler generates the underlying [IL](https://en.wikipedia.org/wiki/Common_Intermediate_Language) code, 
+marking our `Main` method as `async` has the potential to cause problems. As a result, we must pass the program execution to an `async` method.
+
+Head back to your *Program.cs* tab and empty the `Main` method by deleting line 9.
+
+![Code Editor](/images/basics_first_bot_09.png)
+
+Now, create a new `static` method named `MainAsync` beneath your `Main` method. Have it return type `Task` and mark it as `async`.
+After that, add `MainAsync().GetAwaiter().GetResult();` to your `Main` method.
+
+```cs
+static void Main(string[] args)
+{
+    MainAsync().GetAwaiter().GetResult();
+}
+
+static async Task MainAsync()
+{
+    
+}	
+```
+
+If you typed this in by hand, Intellisense should have generated the required `using` directive for you.<br/>
+However, if you copy-pasted the snippet above, VS will complain about being unable to find the `Task` type.
+
+Hover over `Task` with your mouse and click on `Show potential fixes` from the tooltip.
+
+![Error Tooltip](/images/basics_first_bot_10.png)
+
+Then apply the recommended solution.
+
+![Solution Menu](/images/basics_first_bot_11.png)
+
+<br/>
+We'll now create a new `DiscordClient` instance in our brand new asynchronous method.
+
+Create a new variable in `MainAsync` and assign it a new `DiscordClient` instance, then pass an instance of `DiscordConfiguration` to its constructor.
+Create an object initializer for `DiscordConfiguration` and populate the `Token` property with your bot token then set the `TokenType` property to `TokenType.Bot`.
+
+```cs
+var discord = new DiscordClient(new DiscordConfiguration()
+{
+    Token = "My First Token",
+    TokenType = TokenType.Bot       
+});
+```
+
+ >[!WARNING]
+ > We hard-code the token in the above snippet to keep things simple and easy to understand.
+ >
+ > Hard-coding your token is *not* a smart idea, especially if you plan on distributing your source code.
+ > Instead you should store your token in an external medium, such as a configuration file or environment variable, and read that into your program to be used with DSharpPlus.
```

Comment:
Interesting idea here: perhaps a short discussion (or an article) of various ways of storing the token, such as files (incl. JSON), environment, etc?